### PR TITLE
Support nebula-spark reader v1.0

### PR DESCRIPTION
**Pull request by Nicole00**
### What changes were proposed in this pull request?
This PR supports spark reading function from nebula, one label for one reading operation.
The reading result is a DataSet for one tag or one edge, including cols configured in options 'returnCols'. If "returnCols" is null or blank, then reading result will include all cols by default.

How to use this function? to see com.vesoft.nebula.example.NebulaReaderExample

### Why are the changes needed?
new feature.
### How was this patch tested?
example test.

### CODE REVIEW

File: tools/nebula-spark/src/main/java/com/vesoft/nebula/bean/ConnectInfo.java [hunk 1]
```diff
@@ -38,13 +34,4 @@ public String getSpaceName() {
         }
         return hostAndPortList;
     }
-
-    @Override
-    public void check() throws IllegalArgumentException {
-        Preconditions.checkArgument(StringUtils.isNotEmpty(spaceName),
```
Comment (darionyaphet) [inline tools/nebula-spark/src/main/java/com/vesoft/nebula/bean/ConnectInfo.java:44]: maybe we can move the check stage into constructor

File: tools/nebula-spark/dependency-reduced-pom.xml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  <parent>
+    <artifactId>tools</artifactId>
+    <groupId>com.vesoft</groupId>
+    <version>1.0.1</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>nebula-spark</artifactId>
+  <build>
+    <plugins>
+      <plugin>
```
Comment (darionyaphet) [inline tools/nebula-spark/dependency-reduced-pom.xml:12]: AFAIK `dependency-reduced-pom.xml` is create by maven?

**Comment by darionyaphet** [comment]
Really Good Start ！ Please Go On ~

File: tools/nebula-spark/src/main/java/com/vesoft/nebula/example/NebulaReaderExample.java [hunk 3]
```diff
@@ -32,29 +32,30 @@ public static void main(String[] args) {
         Dataset<Row> vertexDataset = sparkSession
                 .read()
                 .format("com.vesoft.nebula.reader.NebulaDataSource")
-                .option("importType", Type.VERTEX.getType())
+                .option("scanType", Type.VERTEX.getType())
```
Comment (darionyaphet) [inline tools/nebula-spark/src/main/java/com/vesoft/nebula/example/NebulaReaderExample.java:35]: type maybe better ?

File: tools/nebula-spark/src/main/java/com/vesoft/nebula/example/NebulaReaderExample.java [hunk 4]
```diff
@@ -32,29 +32,30 @@ public static void main(String[] args) {
...
-                .option("returnCols", "like=likeness;select=grade")
+                .option("spaceName", "nb")
+                .option("label", "follow")
+                .option("returnCols", "degree")
                 .load();
         LOGGER.info("edge schema: ");
         edgeDataset.printSchema();
         edgeDataset.show();
-
+//
```
Comment (darionyaphet) [inline tools/nebula-spark/src/main/java/com/vesoft/nebula/example/NebulaReaderExample.java:57]: remove it ?

File: tools/nebula-spark/src/main/java/com/vesoft/nebula/bean/ConnectInfo.java (hunk 1 above)
Comment (Nicole00) [reply tools/nebula-spark/src/main/java/com/vesoft/nebula/bean/ConnectInfo.java:44]: The check stage has been moved to NebulaDataSource.createRelation，if it's better to check the parameters in ConnectInfo's constructor and ScanInfo's constructor?

File: tools/nebula-spark/dependency-reduced-pom.xml (hunk 2 above)
Comment (Nicole00) [reply tools/nebula-spark/dependency-reduced-pom.xml:12]: yeah，I should remove this file.

File: tools/nebula-spark/src/main/java/com/vesoft/nebula/example/NebulaReaderExample.java (hunk 4 above)
Comment (Nicole00) [reply tools/nebula-spark/src/main/java/com/vesoft/nebula/example/NebulaReaderExample.java:57]: 👌

File: tools/nebula-spark/src/main/java/com/vesoft/nebula/example/NebulaReaderExample.java (hunk 3 above)
Comment (Nicole00) [reply tools/nebula-spark/src/main/java/com/vesoft/nebula/example/NebulaReaderExample.java:35]: I agree.

File: .github/PULL_REQUEST_TEMPLATE [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<!--
+Thanks for sending a pull request!
+-->
```
Comment (darionyaphet) [inline .github/PULL_REQUEST_TEMPLATE:3]: maybe you can find it under .github/workflows/ ...

File: tools/nebula-spark/src/main/java/com/vesoft/nebula/bean/Parameters.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.vesoft.nebula.bean;
+
+public class Parameters {
+    public static final String SPACE_NAME = "spaceName";
```
Comment (darionyaphet) [inline tools/nebula-spark/src/main/java/com/vesoft/nebula/bean/Parameters.java:4]: Great projects have a short name. Such as MySQL Linux and Hadoop

Comment (Nicole00) [reply tools/nebula-spark/src/main/java/com/vesoft/nebula/bean/Parameters.java:4]: Great! The dataSource now has a shortName "nebula",  you can read nebula data by SparkSession.read().format("nebula").option......

File: .github/PULL_REQUEST_TEMPLATE (hunk 5 above)
Comment (Nicole00) [reply .github/PULL_REQUEST_TEMPLATE]: OK, the template is deleted now.

File: tools/nebula-spark/pom.xml [hunk 7]
```diff
@@ -21,17 +21,12 @@
     <dependencies>
         <dependency>
             <groupId>org.apache.spark</groupId>
-            <artifactId>spark-core_2.12</artifactId>
+            <artifactId>spark-core_2.11</artifactId>
             <version>${spark.version}</version>
         </dependency>
         <dependency>
             <groupId>org.apache.spark</groupId>
-            <artifactId>spark-graphx_2.12</artifactId>
```
Comment (darionyaphet) [inline tools/nebula-spark/pom.xml:29]: `graphx` will support later so don't remove it

File: tools/nebula-spark/src/main/java/com/vesoft/nebula/reader/DefaultSource.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        // check and parse parameter returnCols
+        Preconditions.checkArgument(parameters.get(Parameters.RETURN_COLS).isDefined(), "returnCols is not configured.");
+        String returnCols = parameters.get(Parameters.RETURN_COLS).get();
+        boolean isReturnColLegal = StringUtils.isBlank(returnCols) || Pattern.matches(RETURN_COL_REGEX, returnCols);
+        Preconditions.checkArgument(isReturnColLegal,
+                "returnCols '%s' is illegal, the pattern should be blank or string like a,b",
+                returnCols);
+
+        // check and parse parameter partitionNumber
+        String partitionNumber = parameters.getOrElse(Parameters.PARTITION_NUMBER,
```
Comment (darionyaphet) [inline tools/nebula-spark/src/main/java/com/vesoft/nebula/reader/DefaultSource.java:69]: if not found throw an error

File: tools/nebula-spark/pom.xml (hunk 7 above)
Comment (Nicole00) [reply tools/nebula-spark/pom.xml:29]: graphx is not removed,  I modified its version to 2.11.

File: tools/nebula-spark/src/main/java/com/vesoft/nebula/reader/DefaultSource.java (hunk 8 above)
Comment (Nicole00) [reply tools/nebula-spark/src/main/java/com/vesoft/nebula/reader/DefaultSource.java:69]: ok.

**Review by darionyaphet — APPROVED** [review]

**Pull request merged**